libs/wal_decoder/src/models.rs

```diff
@@ -45,10 +45,9 @@ pub struct InterpretedWalRecord {
     /// Optional metadata record - may cause writes to metadata keys
     /// in the storage engine
     pub metadata_record: Option<MetadataRecord>,
-    /// Images or deltas for blocks modified in the original WAL record.
-    /// The [`Value`] is optional to avoid sending superfluous data to
-    /// shard 0 for relation size tracking.
-    pub blocks: Vec<(CompactKey, Option<Value>)>,
+    /// A pre-serialized batch along with the required metadata for ingestion
+    /// by the pageserver
+    pub batch: SerializedValueBatch,
```

Comment:
nit: consider updating the module doc comment with this.

Reply:
https://github.com/neondatabase/neon/pull/9579/commits/d99e9833abaf2783ecca8b132ab36f184130fd60